proof/crefine/AARCH64/Fastpath_C.thy

```diff
@@ -228,20 +224,19 @@ lemma lookup_fp_ccorres':
    apply (rule ccorres_symb_exec_r)
      apply (rule_tac xf'=ret__int_' in ccorres_abstract, ceqv)
      apply (rule_tac P="rv' = from_bool (isCNodeCap cap)" in ccorres_gen_asm2)
-     apply (simp add: from_bool_0 del: Collect_const cong: call_ignore_cong)
+     apply (simp add: from_bool_0 del: Collect_const)
      apply (rule ccorres_Cond_rhs_Seq)
-      apply (simp add: resolveAddressBits.simps split_def del: Collect_const
-                      split del: if_split)
+      apply (simp add: resolveAddressBits.simps split_def del: Collect_const)
       apply (rule ccorres_drop_cutMon)
       apply (rule ccorres_from_vcg_split_throws[where P=\<top> and P'=UNIV])
        apply vcg
       apply (rule conseqPre, vcg)
       apply (clarsimp simp: throwError_def return_def isRight_def isLeft_def
                             ccap_relation_NullCap_iff)
-     apply (clarsimp simp del: Collect_const cong: call_ignore_cong)
+     apply (clarsimp simp del: Collect_const)
      apply (rule_tac P="valid_cap' cap and valid_objs'"
-                and P'="UNIV \<inter> {s. ccap_relation cap (cap_' s) \<and> isCNodeCap cap}
-                             \<inter> {s. bits_' s = 32 - of_nat bits \<and> bits \<le> 32 \<and> bits \<noteq> 0}"
+                and P'="{s. ccap_relation cap (cap_' s) \<and> isCNodeCap cap}
+                        \<inter> {s. bits_' s = 64 - of_nat bits \<and> bits \<le> 64 \<and> bits \<noteq> 0}" (* FIXME AARCH64 64 *)
```

Comment:
Is there any chance of making the fast path proofs generic over 32/64 bit or should we in this case just not worry about those specific kind of magic numbers? (specifically 32 and 64)

Reply:
Looks like writing things out makes them clearer: yes, I think it is worth getting rid of the magic numbers, because it might at least make some parts of the fast path proofs textually equal or more similar, which makes it easier to apply simple-minded patches like renaming etc.

Reply:
I think making them generic in 32/64 is not feasible, but obvious mentions of machine word lengths are something we should strive for, hence the FIXME. The issue is, if you want this merged ASAP, do I go doing these fixes now or after?

Reply:
As long as the tag is there it's fine to do later.

Sorry for the confusing comment: initially I was thinking we could just remove the tag, because there is no point anyway, but then I discovered what the point is. So everything is fine and no action required for the PR.